src/6_branch.js

```diff
@@ -341,6 +341,11 @@ Branch.prototype['init'] = wrap(
 			utils.cleanApplicationAndSessionStorage(self);
 		}
 
+		// initialize identity_id from storage
+		// note the previous line scrubs this if tracking disabled.
+		var localData = session.get(self._storage, true);
```

Comment:
The second param, bool, leads us to retrieve this from the `branch_session_first`. What happens if I want to change the identity for a session? i.e. first time I call `setIdentity("identityA")`. What if two days latest I want to call ``setIdentity("identityB")` which should yield a different (new?) `identity_id`.

Would we send the new `identityB`?

Reply:
Yes. There is in fact a unit test for it added in this PR. On success, `setIdentity` updates local storage with the new identity and identity_id.